go/pkg/yolosvc/driver_github.go

```diff
@@ -345,23 +413,41 @@ func (worker *githubWorker) batchFromWorkflowRunArtifact(run *github.WorkflowRun
 	return batch
 }
 
-func (worker *githubWorker) batchFromWorkflowRun(run *github.WorkflowRun, prs []*github.PullRequest) *yolopb.Batch {
+func (worker *githubWorker) batchFromWorkflowRun(run *github.WorkflowRun, prs []*github.PullRequest, overrideRun *yolopb.OverrideBuild) *yolopb.Batch {
 	batch := yolopb.NewBatch()
 	createdAt := run.GetCreatedAt().Time
 	updatedAt := run.GetUpdatedAt().Time
 	commitURL := run.GetHeadRepository().GetHTMLURL() + "/commit/" + run.GetHeadSHA()
 
 	newBuild := yolopb.Build{
-		ID:           run.GetHTMLURL(),
-		ShortID:      fmt.Sprintf("%d", run.GetID()),
-		CreatedAt:    &createdAt,
-		UpdatedAt:    &updatedAt,
-		HasCommitID:  run.GetHeadSHA(),
-		Branch:       run.GetHeadBranch(),
-		Driver:       yolopb.Driver_GitHub,
-		CommitURL:    commitURL,
-		HasProjectID: run.GetRepository().GetHTMLURL(),
-		Message:      run.GetHeadCommit().GetMessage(),
+		ID:              run.GetHTMLURL(),
+		ShortID:         fmt.Sprintf("%d", run.GetID()),
+		CreatedAt:       &createdAt,
+		UpdatedAt:       &updatedAt,
+		HasRawCommitID:  run.GetHeadSHA(),
+		HasCommitID:     run.GetHeadSHA(),
+		RawBranch:       run.GetHeadBranch(),
+		Branch:          run.GetHeadBranch(),
+		Driver:          yolopb.Driver_GitHub,
+		CommitURL:       commitURL,
+		HasRawProjectID: run.GetRepository().GetHTMLURL(),
+		HasProjectID:    run.GetRepository().GetHTMLURL(),
+		Message:         run.GetHeadCommit().GetMessage(),
+	}
+
+	if overrideRun != nil {
```

Comment:
here, you can remarshal and unmarhsal the overrideRun against newBuild.

the flow `json | unjson` is clearly unoptimized, but in a way we don't care; and it will bring the security of passing only whitelisted fields IF they are present in the yolo.json file.

Reply:
@moul https://github.com/Doozers/yolo/blob/dev/isma/yolo.json-one-pb/go/pkg/yolosvc/driver_github.go#L439 it isn't an easier way to do it ?

Reply:
I wasn't aware of that method, and I love it.